entry-generation/godot-entry-generator/src/main/kotlin/godot/entrygenerator/generator/provider/RegistrationValuesHandler.kt

```diff
@@ -16,9 +20,26 @@ abstract class RegistrationValuesHandler(
     val propertyDescriptor: PropertyDescriptor,
     val bindingContext: BindingContext
 ) {
+    internal val propertyHintAnnotation = propertyDescriptor.getPropertyHintAnnotation()
+
     abstract fun getPropertyTypeHint(): ClassName
     abstract fun getHintString(): String
 
+    init {
+        checkHintAnnotationUsage()
```

Comment:
Something for later, but we need to check that the annotation hints used for a property makes sense.

Reply:
Wdym?

Reply:
I think @raniejade means we should not be able to annotate with type hint annotations on properties that cannot have this type hint.

Reply:
That's already handled through the `getPropertyTypeHint` and the`getHintString? methods in each `RegistrationValueHandler`